File: test/fixtures/java-project/src/AuthService.java
import java.util.HashSet;
import java.util.Set;
import static java.util.Objects.requireNonNull;

/**
 * Service responsible for managing user authentication sessions.
 * Provides methods for login, logout, and session validation.
 */
public class AuthService implements IAuthenticator {

    private static final int MAX_SESSIONS = 1000;

    private final Set<String> activeSessions = new HashSet<>();

    /**
     * Authenticate a user with the provided credentials.
     * Returns true if authentication succeeds.
     */
    public boolean authenticate(String username, String password) {
        requireNonNull(username, "username must not be null");
        return !username.isEmpty() && !password.isEmpty();
    }

    /**
     * Start a new session for the given token.
     */
    public void startSession(String token) {
        if (activeSessions.size() >= MAX_SESSIONS) {
            throw new IllegalStateException("Max sessions reached");
        }
        activeSessions.add(token);
    }

    /**
     * Check if a session token is currently active.
     */
    public boolean isActive(String token) {
        return activeSessions.contains(token);
    }

    /**
     * End a session by removing its token.
     */
    public void endSession(String token) {
        activeSessions.remove(token);
    }

    /** Private helper — should not be indexed. */
    private void cleanup() {
        activeSessions.clear();
    }
}

File: test/fixtures/csharp-project/src/AuthService.cs
using System;
using System.Collections.Generic;
using static System.Console;

namespace PureContext.Fixture
{
    /// <summary>
    /// Manages user authentication sessions.
    /// </summary>
    public class AuthService : IAuthenticator
    {
        private readonly HashSet<string> _sessions = new();

        /// <summary>Maximum number of concurrent sessions allowed.</summary>
        public const int MaxSessions = 1000;

        /// <summary>
        /// Authenticate a user with username and password.
        /// </summary>
        public bool Authenticate(string username, string password)
        {
            if (string.IsNullOrEmpty(username)) throw new ArgumentNullException(nameof(username));
            return !string.IsNullOrEmpty(password);
        }

        /// <summary>Start a new authenticated session.</summary>
        public void StartSession(string token)
        {
            _sessions.Add(token);
        }

        /// <summary>Check if a session is currently active.</summary>
        public bool IsActive(string token)
        {
            return _sessions.Contains(token);
        }

        /// <summary>End a session by removing its token.</summary>
        public void EndSession(string token)
        {
            _sessions.Remove(token);
        }

        private void Cleanup()
        {
            _sessions.Clear();
        }
    }
}
